plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt

```diff
@@ -410,6 +435,12 @@ private fun recordAddConnection(
     }
 }
 
+// Store network error state in IDE persistent storage to maintain across restarts
```

Comment:
why would we want this?

Reply:
added that while trying to figure out race condition but I think the synchronize lock works better.